### ISSUE TITLE
Improve Markdown support - inline code in lists.

### ISSUE THREAD

**Issue by igormironchik** [doxygen/doxygen#10892]
**Describe the bug**

Inline code in lists is not parsed.

**To Reproduce**

```md
* List with `inline code` does not work.
```

**Expected behavior**

`inline code` in lists should be rendered as inline code.

**Version**

```
1.12.0 (3d494140a674a8344c165a559d841166bbf3b80f*)
```

```
No LSB modules are available.
Distributor ID: Neon
Description:    KDE neon 6.0
Release:        22.04
Codename:       jammy
```

I've built `Doxygen` from sources.

**Comment by albert-github** [comment]
With the little example given I'm nor able reproduce your problem (I do seen the part between the backticks as inline code i.e. in a different font), maybe you expectations are different or the example is insufficient, so:
- please explain what you see and what you expected
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

**Comment by igormironchik** [comment]
Yep, the issue is a little different then I described. Attached an example.

[Doxygen.zip](https://github.com/doxygen/doxygen/files/15436229/Doxygen.zip)

**Comment by albert-github** [comment]
Small analyses:
The line:
```
* [Header with `code`.](#header-with-code)
```
is translated by doxygen into:
```
* @ref "header-with-code" "Header with `code`."
```
for further processing.
We now see that the text between the square brackets is between double quotes, meaning that the backticks in the text have to be taken literal.
Maybe some markdown processing should take place on the string between the square brackets.
Special attention should be payed to links line `[@user]` and `[#10892]` as indicated in the proposed pull request #7339

Note when using:
```
* [Header with <tt>code</tt>.](#header-with-code)
```
the link is shown as expected.

@doxygen what do you think?

**Comment by doxygen** [comment]
@albert-github Since we allow inline code in section titles, I think we should also allow it in references and treat the quoted part of `@ref` as an argument of the command, rather than as a literal text fragment.

Note that also this works correctly
```
@ref anc

# Another Header with `code`. {#anc}
```
so it is a bit inconsistent that a user defined text cannot have inline code, while an automatically generated one can.
